go/vt/vtgate/planbuilder/dml.go

```diff
@@ -184,6 +226,17 @@ func getDMLRouting(where *sqlparser.Where, route *engine.Route) error {
 	return errors.New("unsupported: multi-shard where clause in DML")
 }
 
+// extractValueFromUpdate given an UpdateExpr attempts to extracts the Value
+// it's holding. At the moment it only supports: StrVal, HexVal, IntVal, ValArg.
+// If a complex expression is provided (e.g set name = name + 1), the update will be rejected.
+func extractValueFromUpdate(upd *sqlparser.UpdateExpr, col sqlparser.ColIdent) (pv sqltypes.PlanValue, err error) {
+	if !sqlparser.IsValue(upd.Expr) {
+		err := vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "unsupported: Only values are supported. Invalid update on column: %v", upd.Name.Name)
+		return sqltypes.PlanValue{}, err
+	}
+	return sqlparser.NewPlanValue(upd.Expr)
```

Comment:
The `IsValue` check before building NewPlanValue (which also performs similar checks) has been bothering me. The main reason we have to do it is to exclude list values, which NewPlanValue allows.

I'm thinking that we should (later) build a `NewSinglePlanValue` function and fold the IsValue check into it.

Reply:
Coo, I will make note to myself to do this refactor.